tests/ui/wf/ice-hir-wf-check-anon-const-issue-122989.rs

```diff
@@ -0,0 +1,42 @@
+// Regression test for ICE #122989
+trait Traitor<const N: N<2> = 1, const N: N<2> = N> {
```

Comment:
Can you minimize this at all? There seems to be a significant number of unrelated errors that really hides the underlying issue here.

Reply:
Someone posted a simpler reproducer on the original issue

Reply:
Changed it to the suggested simpler reproducer